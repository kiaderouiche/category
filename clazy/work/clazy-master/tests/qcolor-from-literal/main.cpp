#include <QtGui/QColor>

extern void takingColor(QColor);

void test()
{
    QColor c1; // OK
    QColor c2("#001020"); // Warning
    QColor c3(QString("#001020")); // Warning. TODO
    QColor c4(QLatin1String("#001020")); // Warning. TODO
    QColor c5("#023"); // Warning
    QColor c6("#00112233"); // Warning
    QColor c7("#000011112222"); // Warning
    c4.setNamedColor("red"); // OK
    c4.setNamedColor("#001020"); // Warning
    c4.setNamedColor(QLatin1String("#001020")); // Warning
    takingColor("#001122");  // Warning
}
