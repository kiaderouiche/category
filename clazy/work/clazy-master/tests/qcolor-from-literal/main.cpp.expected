qcolor-from-literal/main.cpp:15:22: warning: The ctor taking ints is cheaper than QColor::setNamedColor(QString) [-Wclazy-qcolor-from-literal]
qcolor-from-literal/main.cpp:16:36: warning: The ctor taking ints is cheaper than QColor::setNamedColor(QString) [-Wclazy-qcolor-from-literal]
qcolor-from-literal/main.cpp:8:15: warning: The QColor ctor taking ints is cheaper than the one taking string literals [-Wclazy-qcolor-from-literal]
qcolor-from-literal/main.cpp:11:15: warning: The QColor ctor taking ints is cheaper than the one taking string literals [-Wclazy-qcolor-from-literal]
qcolor-from-literal/main.cpp:12:15: warning: The QColor ctor taking ints is cheaper than the one taking string literals [-Wclazy-qcolor-from-literal]
qcolor-from-literal/main.cpp:13:15: warning: The QColor ctor taking ints is cheaper than the one taking string literals [-Wclazy-qcolor-from-literal]
qcolor-from-literal/main.cpp:17:17: warning: The QColor ctor taking ints is cheaper than the one taking string literals [-Wclazy-qcolor-from-literal]
